### PR TITLE
chore: pointer update, memory deferencing

### PR DESCRIPTION
**Pull request by SantiiRepair**

### CODE REVIEW

File: autostart_windows.go [hunk 1]
```diff
@@ -4,18 +4,19 @@ package autostart
 /*
 #define WIN32_LEAN_AND_MEAN
 #include <stdint.h>
+#include <stdlib.h>
```
Comment (emersion) [inline autostart_windows.go:7]: This shouldn't be needed.

File: autostart_windows.go [hunk 2]
```diff
@@ -40,10 +41,19 @@ func (a *App) Enable() error {
 	if err := os.MkdirAll(startupDir, 0777); err != nil {
 		return err
 	}
+
+	cShortcutPath := C.CString(a.path())
+	defer C.free(unsafe.Pointer(cShortcutPath))
+	cPath := C.CString(path)
+	defer C.free(unsafe.Pointer(cPath))
+	cArgs := C.CString(args)
+	defer C.free(unsafe.Pointer(cArgs))
```
Comment (emersion) [inline autostart_windows.go:50]: These aren't actually used in the call below.

File: autostart_windows.c [hunk 3]
```diff
@@ -1,47 +1,53 @@
...
 #include <string.h>
 #include <objbase.h>
-#include <shlobj.h>
+#include <windows.h>
 
-uint64_t CreateShortcut(char *shortcutA, char *path, char *args) {
-	IShellLink*   pISL;
-	IPersistFile* pIPF;
-	HRESULT       hr;
+uint64_t CreateShortcut(char *s, char *path, char *args)
```
Comment (emersion) [inline autostart_windows.c:9]: Seems like the only changes here are code style?

If you want to make code style changes, can you split these in a separate file? (Note, note sure I like all of these changes.)

Comment (SantiiRepair) [reply autostart_windows.c:9]: Alright

**Pull request closed without merging**